src/Spd.Manager.Licence/BizProfileManager.cs

```diff
@@ -182,4 +196,28 @@ private async Task<PortalUserResp> AddPortalUserToBiz(BceidIdentityInfo info, Gu
             PortalUserServiceCategory = PortalUserServiceCategoryEnum.Licensing
         }, ct);
     }
+
+    private async Task ProcessBranchAddresses(List<BranchAddr> branches, List<BranchAddr> branchesToProcess, Guid bizId, CancellationToken ct)
+    {
+        // Remove branches defined in the entity that are not part of the request
+        var modifiedBranches = branchesToProcess.Where(b => b.BranchId != Guid.Empty && b.BranchId != null);
+        List<Guid?> addressesToRemove = branches.Where(b => modifiedBranches.All(mb => mb.BranchId != b.BranchId)).Select(b => b.BranchId).ToList();
+        await _addressRepository.DeleteAddressesAsync(addressesToRemove, ct);
+
+        // Update branches
+        UpsertAddressCmd updateAddressCmd = new()
+        {
+            Addresses = modifiedBranches
+        };
+        await _addressRepository.UpdateAddressesAsync(updateAddressCmd, ct);
+
+        // Create branches
+        List<BranchAddr> addressesToCreate = branchesToProcess.Where(b => b.BranchId == Guid.Empty || b.BranchId == null).ToList();
+        UpsertAddressCmd createAddressCmd = new()
+        {
+            BizId = bizId,
+            Addresses = addressesToCreate
+        };
+        await _addressRepository.CreateAddressesAsync(createAddressCmd, ct);

```

Comment:
all this needs 3 transactions. I will create an improvement ticket as Update an array. All delete, update and create are done in the repository layer with one transaction.
It should apply to both Alias and Address.